src/Application/Items/Queries/GetItemsByBaseIdQuery.cs

```diff
@@ -0,0 +1,35 @@
+using AutoMapper;
+using Crpg.Application.Common.Interfaces;
+using Crpg.Application.Common.Mediator;
+using Crpg.Application.Common.Results;
+using Crpg.Application.Items.Models;
+using Microsoft.EntityFrameworkCore;
+
+namespace Crpg.Application.Items.Queries;
+
+public record GetItemsByBaseIdQuery : IMediatorRequest<IList<ItemViewModel>>
+{
+    public string BaseId { get; init; } = string.Empty;
+    internal class Handler : IMediatorRequestHandler<GetItemsByBaseIdQuery, IList<ItemViewModel>>
+    {
+        private readonly ICrpgDbContext _db;
+        private readonly IMapper _mapper;
+
+        public Handler(ICrpgDbContext db, IMapper mapper)
+        {
+            _db = db;
+            _mapper = mapper;
+        }
+
+        public async Task<Result<IList<ItemViewModel>>> Handle(GetItemsByBaseIdQuery req, CancellationToken cancellationToken)
+        {
+            var items = await _db.Items
+                .AsNoTracking()
```

Comment:
You can use ProjectTo method here to create a query that can be directly mapped to an ItemViewModel. You'll find some usages in the code base.

Reply:
I used the Restriction Query as an example and it makes the test not compile : 
![image](https://github.com/verdie-g/crpg/assets/47090987/934bfea4-1595-45b2-8925-a8a961d27a69)

Reply:
@verdie-g  this part of the itemviewmodel cannot be projected into an sql query : 
.ForMember(i => i.Weapons, opt => opt.MapFrom(i => i.GetWeapons()));
"
Looking at your ItemViewModel class and the mapping function, the problem appears to be happening because of this line:

csharp

.ForMember(i => i.Weapons, opt => opt.MapFrom(i => i.GetWeapons()));

This is where the complex LINQ query is coming from that Entity Framework can't translate into SQL. In this case, the GetWeapons function is likely producing a result that cannot be translated into a SQL statement."